task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue

```diff
@@ -303,9 +305,43 @@ export default {
                             this.menuActions = this.menuActions.filter(menuAction => menuAction.enabled);
                         });
                     }
+                } else if ( event.detail.name === 'noProject' ) {
+                    this.menuActions = [];
+                    this.enableDelete = event.detail.task.createdBy === eXo.env.portal.userName;
+                    this.enableClone = event.detail.task.assignee === eXo.env.portal.userName || event.detail.task.coworker.includes(eXo.env.portal.userName);
+                    this.addMenuAction(this.$t('label.delete'), 'uiIconTrash', this.enableDelete, 'deleteTask');
+                    this.addMenuAction(this.$t('label.clone'), 'uiIconCloneNode', this.enableClone, 'cloneTask');
+                    this.menuActions = this.menuActions.filter(menuAction => menuAction.enabled);
                 }
             }
         });
+       const urlPath = document.location.pathname
```

Comment:
You already have the task loaded inside the drawer, why do you need to reimport it based on the URL params ?

Reply:
because we need to show the three dots menu when the task is opened directly by URL, actually we add an event with the open drawer event to show the three dots menu

Reply:
Couldn't we add a listener that will receive the task opening by URL and use it to check which actions will be displayed. I don't find a good reason to read from URL

Reply:
@SaraBoutej Could you confirm that there is no other way than reading the current page's URL  to get the displayed task ID ?

Reply:
@ahamdi yes we can, i used watch!